app/Models/Post.php

```diff
@@ -31,4 +31,36 @@ public function isApproved(): bool
     {
         return !is_null($this->approved_at);
     }
+
+    public function scopeTitle($query, $title)
+    {
+        if ($title)
+            return $query->where('title', 'LIKE', "%$title%");
+    }
```

Comment:
Una de las cosas que nunca nos dijeron en la universidad, es que cuando escribimos código es más importante que este sea muy legible y fácil de entender para los humanos, que para la computadora.

Un tip muy usado es uno que llaman el Camino Felíz. Esto ayuda a que el código sea más fácil de entender. 

Consiste en evaluar primero los casos en los cuáles no se debe ejecutar el código. Si se presentan esos casos se retorna sin hacer nada, dejando al final la ejecución de la lógica del método.

Esto ayuda para reducir en la mayoría de los casos if-elses anidados que dificultan la lectura. Siempre debemos eviar utilizar los else's

Revisa esta guía https://spatie.be/guidelines/laravel-php#content-happy-path

En este caso, podemos preguntar primero si la variable `$value` está vacía, en ese caso retornamos el `$query` sin aplicarle ninguna condicional.

Si no entra a la condición anterior, quiere decir que debemos ejecutar el condicional y retornarlo, el camino felíz.

Este método puede quedar así:
```php
public function scopeTitle(Builder $query, string $title): Builder
{
    if (empty($title)) {
        return $query;
    }

    return $query->where('title', 'LIKE', "%$title%");
}
```

Recuerda indicar los tipos, tanto en los parámetros como el retorno del método.

Para la variable `$query` y el retorno del método el tipo es `Illuminate\Database\Eloquent\Builder`, es una clase de Laravel. Para el `$value` el tip0 es `string`